torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1108,7 +1108,8 @@ bool ProcessGroupNCCL::dumpDebuggingInfo() {
   // Serialize all calls to this function to avoid corrupting data, but allow
   // multiple calls in one runtime. User is responsible for preserving the
   // output file from an earlier call before a later call overwrites it.
-  std::lock_guard<std::mutex> lock(writeDebugInfoMutex_);
+  static std::mutex writeDebugInfoMutex;
```

Comment:
wow, for my learning purpose, does it now acts like a global variable so that it works across different PG objects?

Reply:
well, as far as lifetime it gets created on the first usage, and lives forever.  Subsequent times that line of code just shares the previously created one.  But as far as scope, its only visible inside that {} region.